Microsoft.Azure.Cosmos/src/GatewayAccountReader.cs

```diff
@@ -64,6 +64,10 @@ private async Task<AccountProperties> GetDatabaseAccountAsync(Uri serviceEndpoin
                     }
                 }
             }
+            catch (ObjectDisposedException) when (this.cancellationToken.IsCancellationRequested)
+            {
+                throw new OperationCanceledException("Client is being disposed, cancelling further operations.");
```

Comment:
nit: add more info to the exception message to make it easier to troubleshoot when there are multi tenant scenarios.
```suggestion
                throw new OperationCanceledException($"Client is being disposed for {serviceEndpoint} at {DateTime.UtcNow}, cancelling further operations.");
```